### PR TITLE
updating TS branch for takehome

### PR DESCRIPTION
**Pull request by funador**
Updates the typescript branch to have take home instructions.

### CODE REVIEW

**Review by seanmooney-extend — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,46 +1,20 @@
...
 
-In the project directory, you can run:
+- Inside this folder run `yarn` and then `yarn start` and `yarn test`, [help setting up yarn](https://classic.yarnpkg.com/en/docs/install).
+- CRA requires `Node 14.0.0 or later version`, instructions to [change your node version](https://stackoverflow.com/questions/53785383/how-to-change-node-version-with-nvm/53785482).
+- Create a new private repo on Github and follow the instructions to [push your local code to your remote repository](https://docs.github.com/en/get-started/using-git/pushing-commits-to-a-remote-repository).
+- Check out a new branch locally to complete the challenge
+- When you are ready, push that branch and open a PR
+- Add `funador` as a collaborator on your private repo
+- Any questions or ready for a review? Please send an email to `jesse@extend.com`
+- If you prefer JavaScript, use the [JavaScript branch](https://github.com/helloextend/frontend-challenge-scaffold/tree/typescript) as your starter.
```
Comment (jcast90) [inline README.md]: Should it be this link?
```suggestion
- If you prefer JavaScript, use the [JavaScript branch](https://github.com/helloextend/frontend-challenge-scaffold/tree/javascript) as your starter.
```

File: README.md [hunk 2]
```diff
@@ -1,46 +1,20 @@
...
-You don’t have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.
-
-## Learn More
-
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
-
-To learn React, check out the [React documentation](https://reactjs.org/).
+- Follow the instructions in App.tsx
+- Use any packages that you want
+- Style the UI however you want
```
Comment (jcast90) [inline README.md:20]: could we maybe have it be with where they only use 1 option though. Like don't mix styled components and `scss` modules or something like that? Or maybe we only ever saw the mixed styles due to us including styled components. I feel like for this actually, we want to see that they know css, so maybe ui-kits? Like they need to write some css.

File: src/App.test.tsx [hunk 3]
```diff
@@ -1,9 +1,6 @@
...
+import App from "./App";
 
-test('should chunk an array properly', () => {
-  const testData = Array.from(new Array(55)).map((value, index) => `element ${index}`)
-  const result = chunk(testData, 10)
-  expect(result.length).toEqual(6)
-  expect(result[0].length).toEqual(10)
-  expect(result[5].length).toEqual(5)
+test("a user interaction", () => {
+  const wrapper = render(<App />);
```
Comment (jcast90) [inline src/App.test.tsx:6]: maybe we can add a comment like:
```
// delete this test
```

or maybe more info like:

```
// delete test, we don't need to test that the App renders, but more focused on unit tests
```

Comment (jcast90) [reply src/App.test.tsx:6]: or something like that? or maybe it's not necessary and is part of the test to see how much they know about testing?

File: src/App.tsx [hunk 4]
```diff
@@ -1,21 +1,28 @@
...
+ * @returns { "message": ["https://images.dog.ceo/breeds/hound-afghan/n02088094_1126.jpg", ...] }
+ *
+ * Step 1: Render all the images returned
+ * Step 2: Create chunks of 10 images from the returned list
+ * Step 3: Render the first 10 of the list
+ * Step 4: Build a way to increment to the next 10
+ * Step 5: Build a way to decrement to the previous 10
+ * Step 6: Create a UI that shows which page you're viewing
+ * Step 7: Write a test that covers a user interaction
+ *
```
Comment (jcast90) [inline src/App.tsx:15]: Maybe we could add some bonus points here that we'd kind of expect a senior/higher level of an engineer to complete? like maybe create a way to search, or fetching past 55 to create a kind of fake server pagination, or idk maybe this is enough.

**Review by jcast90 — COMMENTED** [review]
Nice, I like this. I feel like it's more clear than the previous one. Also I feel like it'll be a little simpler/less time consuming for the candidate. But overall this seems like it'll cover a lot of skills we are kind of looking for/expecting from an FE engineer joining the company :) Good work dude!

File: README.md (hunk 2 above)
Comment (jcast90) [reply README.md:20]: Oh also, maybe we specify to not use class based components and for them to make use of hooks. I know we had some candidates in the past that weren't 100% up to speed on recent versions of React. So maybe we specify that here?

File: README.md (hunk 1 above)
Comment (funador) [reply README.md]: Good point, will pull this out.

File: src/App.tsx (hunk 4 above)
Comment (funador) [reply src/App.tsx:15]: I am inclined to stay on the light end here and see how things turn out for the first few.  We should come up with a couple of extension tasks to ask during interviews though and maybe roll them if we feel we need to expand the scope of the take home.

File: README.md (hunk 2 above)
Comment (funador) [reply README.md:20]: I was kind of thinking this open ended and that gives us follow up material in the interview.  I think a good engineer would stick to one css approach, so if there is more than one that would be a flag. 

But why don't we see what happens for the first few and then make adjustments as required?

File: src/App.test.tsx (hunk 3 above)
Comment (funador) [reply src/App.test.tsx:6]: Added a comment

**Review by jcast90 — APPROVED** [review]

**Review by m-schrepel — APPROVED** [review]

**Pull request merged**